### DISCUSSION TITLE
Testing in tournament

### DISCUSSION THREAD

**Discussion by rnz1234** [yasserfarouk/scml#48] [one-shot-qa] [answered]
Hi, several questions:
1. When testing in tournament, I see that although the agent has something to sell/buy (needed by supply/demand) in a lot of cases the system just didn't run the propose method to start negotiation. Why is that ?
2. What are the pre defined agents to test against ?
3. What are the alpha and beta coefficients in penalty equation in the document of oneshot ?
thanks

**Comment by yasserfarouk** [comment, marked as answer]
> When testing in tournament, I see that although the agent has something to sell/buy (needed by supply/demand) in a lot of cases the system just didn't run the propose method to start negotiation. Why is that ?

- If you are speaking about the std/collusion tracks then negotiations must be started by one agent and accepted by the other to start. The simulator controls neither.
- If you are speaking about the oneshot track, then negotiations with all partners should start automatically every step. 

If you see a case in which either of these two rules is broken, it is a bug. Please report it as an issue and give us some way to reproduce it.

> What are the pre defined agents to test against ?

Not sure what you are asking about here. In the official competition, only participant-submitted agents exist in the world. In the online competition, one built-in agent (MarketAwareDecentralizingAgent for std/collusion and GreedyOneshotAgent for oneshot) is added to the world. For your own run tests, you can use any agents in the scml or scml-agents packages

> What are the alpha and beta coefficients in the penalty equation in the document of oneshot ?

These are defined in full detail in the game description [Section 4 Page 6 here](http://www.yasserm.com/scml/scml2021oneshot.pdf). 
In summary:
- $\alpha$ defines a penalty for buying too much (disposal penalty).
- $\beta$ defines a penalty for selling too much (disposal penalty).

If your question is about their values, you can find this information in [Table 3, Page 14 of the same document](http://www.yasserm.com/scml/scml2021oneshot.pdf). These values are for reference only and are subject to be changed in the actual competition.

**Reply by yasserfarouk** [reply]
hi @rnz1234 

Please let me know if the answer was adequate for you. thanks
